code/android-core-library/src/main/java/com/adobe/marketing/mobile/configuration/AppIdManager.kt

```diff
@@ -0,0 +1,91 @@
+/*
+  Copyright 2022 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+ */
+
+package com.adobe.marketing.mobile.configuration
+
+import com.adobe.marketing.mobile.LoggingMode
+import com.adobe.marketing.mobile.MobileCore
+import com.adobe.marketing.mobile.services.DataStoring
+import com.adobe.marketing.mobile.services.DeviceInforming
+
+/**
+ * Manages the storage and retrieval of AEP appID from shared preferences and the app manifest.
+ */
+internal class AppIdManager(
+    private val dataStoreService: DataStoring,
+    private val deviceInfoService: DeviceInforming
+) {
+
+    companion object {
+        private const val LOG_TAG = "AppIdManager"
+    }
+
+    /**
+     * Saves the appId provided into shared preferences.
+     *
+     * @param appId the appId to store/update to shared preferences
+     */
+    internal fun saveAppIdToPersistence(appId: String) {
+        if (appId.isBlank()) {
+            MobileCore.log(
+                LoggingMode.VERBOSE,
+                LOG_TAG,
+                "Attempting to set empty App Id into persistence."
+            )
+            return
+        }
+
+        dataStoreService.getNamedCollection(ConfigurationStateManager.DATASTORE_KEY)
+            ?.setString(ConfigurationStateManager.PERSISTED_APPID, appId)
+    }
+
+    /**
+     * Removes the existing appId stored in shared preferences.
+     */
+    internal fun removeAppIDFromPersistence() {
+        dataStoreService.getNamedCollection(ConfigurationStateManager.DATASTORE_KEY)
+            ?.remove(ConfigurationStateManager.PERSISTED_APPID)
+    }
+
+    /**
+     * Retrieves appId from persistence when available. Falls back to
+     * retrieving it from manifest if nothing is persisted.
+     *
+     * @return the appId stored in manifest or persistence (in that order) if one exists;
+     *         null otherwise
+     */
+    internal fun loadAppId(): String? {
+        return getAppIDFromPersistence() ?: getAppIDFromManifest().also { manifestAppId ->
```

Comment:
I'm thinking of an extreme case when the Android app is updated with a new app id in the manifest file. Should we first load the app id from the manifest?